### ISSUE TITLE
Autocomplete works, but speed is so slow(

### ISSUE THREAD

**Issue by Partysun** [jdonaldson/vaxe#66]
I have more than 0.5-1 sec for some classes/functions lag after write dot symbol.
Thank you,

**Comment by jdonaldson** [comment]
i'm going to close that out, reopen if you see any more issues.

**Comment by slouffka** [comment]
Same problem here. It's insanely slow. It's annoying and getting me mad because I use Vim for its ```SPEED```. Can't work like that :(

Btw YCM for C++ works lightning fast. Why it's so slow for Haxe? All Haxe libs are compiled to C++ every time I type anything? Sorry to say that but it makes Vaxe unusable though you've did really great job with it in total.

It there any way to make the completion fast? What part of my code gets recompiled?

**Comment by jdonaldson** [comment]
Sorry to hear.  It shouldn't slow things down "every time you type something".  Only when you
1) invoke the autocompletion method directly
2) use an autompletion package like YCM that invokes the autocompletion method after a trigger key (period or paren).

Same request goes for the previous person... give me some information about the hxml generated by vaxe, as well as give me some information on which version of openfl/haxe you're using.  A reproducible example would be great as well... does this happen on one of the "demo" projects?

**Comment by slouffka** [comment]
Yeah. You're right about triggering. My fault to say that way. But I type fast so that feels just like every time I type :) I mean now it's times faster to type without autocomplete what I am exactly doing to write code. Sometimes it so slow that my code got glithches which makes editing a pain. I'll make a screenshot when it happens. Occured only with Haxe/Vaxe. Not with any other plugins or languages like C++ or JavaScript.

Yes, it happens with any code. Like Flixel TiledEditor demo for example.

Just tried openfl 3.6.1 and 4.3.1. Used sample DisplayingABitmap. Both versions lags like a 1-2 seconds before showing the list on typing period or invoking with ctrl+space. Not only first time but any time I invoke it with any method.

I noticed that lag on 3 different machines. Core 2 Duo desktop, i3 notebook, i7 desktop. Only i7 works somewhat like okay but even that I can't call fast.

We could look into VS Code Haxe autocompletion implementation because it works fast and feels very smooth. Looks like they use some other way to make it. https://github.com/jcward/vscode-haxe.

**Comment by jdonaldson** [comment]
Thanks, I'll check it out.  I know the folks working with vscode, and I'll see how they're doing it.

**Comment by slouffka** [comment]
Updated a link to vshaxe. First it was to the old extension.

**Comment by bloognoo** [comment]
I know this has been resting a while, and I don't know how much this helps as we're years down the line, but I've had a recent performance drop following updating my plugins that I rectified by knockout out Syntastic and YouCompleteMe.

Syntastic lags buffer swapping ; air/powerline updates maybe 3 seconds before the buffer updates
YouCompleteMe lags when hitting periods, brackets, etc and stalls 3 seconds, including visual corruption to the buffer until ycm returns and the buffer refreshes.

I'm going to have a dig and see if it's the haxe language server that is still bogging down the response, and maybe what changed. This may have been round about the 8.2 of vim which added async capabilities, sothat's another moving part to check :)

If anyone's ahead of me on this, please let me know.
